contracts/treasury/LivepeerGovernor.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/GovernorUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorSettingsUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorTimelockControlUpgradeable.sol";
+
+import "../bonding/libraries/EarningsPool.sol";
+import "../bonding/libraries/EarningsPoolLIP36.sol";
+import "../polling/PollCreator.sol";
+
+import "../ManagerProxyTarget.sol";
+import "../IController.sol";
+import "../rounds/IRoundsManager.sol";
+import "./GovernorCountingOverridable.sol";
+import "./BondingCheckpointsVotes.sol";
+
+/**
+ * @title LivepeerGovernor
+ * @notice Core contract for Livepeer governance, starting as the treasury governor.
+ * @dev If we ever add fields to this class or more extensions, make sure to add a storage gap to our custom
+ * GovernorCountingOverridable extension.
+ */
+contract LivepeerGovernor is
+    Initializable,
+    ManagerProxyTarget,
+    GovernorUpgradeable,
+    GovernorSettingsUpgradeable,
+    GovernorTimelockControlUpgradeable,
+    GovernorVotesUpgradeable,
+    GovernorVotesQuorumFractionUpgradeable,
+    GovernorCountingOverridable
+{
+    /**
+     * @notice TreasuryGovernor constructor. Only invokes constructor of base Manager contract with provided Controller.
+     * @dev This constructor will not initialize any state variables besides `controller`. The `initialize` function
+     * must be called through the proxy after construction to initialize the contract's state in the proxy contract.
+     * @param _controller Address of Controller that this contract will be registered with
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address _controller) Manager(_controller) {
+        _disableInitializers();
+    }
+
+    /**
+     * Initializes the LivepeerGovernor instance. This requires the following contracts to have already been deployed
+     * and registered on the controller:
+     * - "Treasury" (TimelockControllerUpgradeable)
+     * - "BondingCheckpointsVotes"
+     * - "PollCreator"
+     */
+    function initialize() public initializer {
+        __Governor_init("LivepeerGovernor");
+        __GovernorSettings_init(
+            1, /* 1 round/day voting delay */
+            10, /* 10 rounds/days voting period */
+            100e18 /* 100 LPT min proposal threshold */
```

Comment:
Might want to make these `initialize` args instead for configurability per network on deploys?